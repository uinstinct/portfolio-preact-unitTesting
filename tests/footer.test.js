import { fireEvent, screen, waitFor } from '@testing-library/react';

import Footer from '@/containers/Footer';
import { customRender } from './testUtils';
import { footerConstants } from './testUtils/constants';

jest.mock("@/containers/Footer/constants", () => ({
    __esModule: true,
    default: require('./testUtils/constants').footerConstants,
}));

describe("test the footer component and its functionalities |", () => {

    it("should render and match its snapshot", () => {
        const { container } = customRender(<Footer />);
        const link = container.querySelector('a');

        expect(screen.getByText(/nextjs/gi)).toBeInTheDocument();
        expect(link).toHaveAttribute('href', footerConstants[0].link);
        expect(container.innerHTML).toMatchSnapshot();
    });

    it("should match snapshot when dark mode is on", () => {
        const { container } = customRender(<Footer />, null, 'dark');
        expect(container.innerHTML).toMatchSnapshot();
    });

});