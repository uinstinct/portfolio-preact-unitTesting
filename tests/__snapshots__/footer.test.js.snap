// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the footer component and its functionalities | should match snapshot when dark mode is on 1`] = `"<span id=\\"findMe\\" style=\\"visibility: hidden;\\" class=\\"jsx-1216231799\\"></span><footer class=\\"jsx-1216231799 footer dark\\"><div class=\\"jsx-1216231799 contact-container\\"><span class=\\"jsx-1216231799 header\\">Talk With Me</span><div class=\\"jsx-1216231799 contact-icons\\"><a target=\\"_blank\\" rel=\\"noopener noreferrer\\" class=\\"chakra-link css-iqkmuh\\" href=\\"test@gmail.com\\"><svg viewBox=\\"0 0 24 24\\" focusable=\\"false\\" class=\\"chakra-icon css-15u7b0s\\"><g stroke=\\"currentColor\\" stroke-width=\\"1.5\\"><path stroke-linecap=\\"round\\" fill=\\"none\\" d=\\"M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25\\"></path><path fill=\\"currentColor\\" stroke-linecap=\\"round\\" d=\\"M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0\\"></path><circle fill=\\"none\\" stroke-miterlimit=\\"10\\" cx=\\"12\\" cy=\\"12\\" r=\\"11.25\\"></circle></g></svg></a><a target=\\"_blank\\" rel=\\"noopener noreferrer\\" class=\\"chakra-link css-iqkmuh\\" href=\\"apple@icloud.com\\"><svg viewBox=\\"0 0 24 24\\" focusable=\\"false\\" class=\\"chakra-icon css-1hlubd3\\"><g stroke=\\"currentColor\\" stroke-width=\\"1.5\\"><path stroke-linecap=\\"round\\" fill=\\"none\\" d=\\"M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25\\"></path><path fill=\\"currentColor\\" stroke-linecap=\\"round\\" d=\\"M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0\\"></path><circle fill=\\"none\\" stroke-miterlimit=\\"10\\" cx=\\"12\\" cy=\\"12\\" r=\\"11.25\\"></circle></g></svg></a></div></div><div class=\\"jsx-1216231799 built-with\\"><div class=\\"jsx-1216231799 border\\"></div>My Portfolio Site Built With<a target=\\"_blank\\" rel=\\"noopener noreferrer\\" class=\\"chakra-link css-1bvdrhg\\" href=\\"https://github.com/aditya-mitra/aditya-mitra.github.io/\\"><span class=\\"jsx-1216231799 built\\">NextJS</span></a></div></footer>"`;

exports[`test the footer component and its functionalities | should render and match its snapshot 1`] = `"<span id=\\"findMe\\" style=\\"visibility: hidden;\\" class=\\"jsx-1216231799\\"></span><footer class=\\"jsx-1216231799 footer null\\"><div class=\\"jsx-1216231799 contact-container\\"><span class=\\"jsx-1216231799 header\\">Talk With Me</span><div class=\\"jsx-1216231799 contact-icons\\"><a target=\\"_blank\\" rel=\\"noopener noreferrer\\" class=\\"chakra-link css-iqkmuh\\" href=\\"test@gmail.com\\"><svg viewBox=\\"0 0 24 24\\" focusable=\\"false\\" class=\\"chakra-icon css-15u7b0s\\"><g stroke=\\"currentColor\\" stroke-width=\\"1.5\\"><path stroke-linecap=\\"round\\" fill=\\"none\\" d=\\"M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25\\"></path><path fill=\\"currentColor\\" stroke-linecap=\\"round\\" d=\\"M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0\\"></path><circle fill=\\"none\\" stroke-miterlimit=\\"10\\" cx=\\"12\\" cy=\\"12\\" r=\\"11.25\\"></circle></g></svg></a><a target=\\"_blank\\" rel=\\"noopener noreferrer\\" class=\\"chakra-link css-iqkmuh\\" href=\\"apple@icloud.com\\"><svg viewBox=\\"0 0 24 24\\" focusable=\\"false\\" class=\\"chakra-icon css-1hlubd3\\"><g stroke=\\"currentColor\\" stroke-width=\\"1.5\\"><path stroke-linecap=\\"round\\" fill=\\"none\\" d=\\"M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25\\"></path><path fill=\\"currentColor\\" stroke-linecap=\\"round\\" d=\\"M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0\\"></path><circle fill=\\"none\\" stroke-miterlimit=\\"10\\" cx=\\"12\\" cy=\\"12\\" r=\\"11.25\\"></circle></g></svg></a></div></div><div class=\\"jsx-1216231799 built-with\\"><div class=\\"jsx-1216231799 border\\"></div>My Portfolio Site Built With<a target=\\"_blank\\" rel=\\"noopener noreferrer\\" class=\\"chakra-link css-1bvdrhg\\" href=\\"https://github.com/aditya-mitra/aditya-mitra.github.io/\\"><span class=\\"jsx-1216231799 built\\">NextJS</span></a></div></footer>"`;
